### DISCUSSION TITLE
Serial console

### DISCUSSION THREAD

**Discussion by cpyarger** [malcolmholmes/pico-clock-green-python#12] [Ideas]
Personally I would like to hook up the UART to an ESP32 and have the ESP send data over the serial port to set the time via NTP and display messages. among setting other options.  since we have access to serial via the usb, as well as GPIO, it makes sense to utilize that rather than just using the 3 button interface, imo.

**Comment by malcolmholmes** [comment]
You're suggesting hardware changes here? Would be neat to have one of these clocks with wifi/etc, but isn't that a question for the manufacturers? Or am I missing something?

**Comment by cpyarger** [comment]
No there are the doubled pin headers the pico is plugged into. And this
would allow using the usb serial as an interface to send messages and set
time etc as well

**Comment by cpyarger** [comment]
it's not great but I have got a simple get and set time via serial command working, including the script to run on your computer to set the time. 

https://github.com/cpyarger/pico-clock-green-python/tree/serial-api

below is the code to run on your computer, make sure to change your serial port. 
```

import datetime as dt
import serial

# (YY, MM, mday, hh, mm, ss, wday, yday) = t
x=dt.datetime.now().strftime("%Y, %M, %d, %H, %M, %S, %w, %U")

y="set, "
z=y+x
print(z)
ser = serial.Serial('COM10', 115200)
ser.write(str(z).encode('utf-8'))
ser.write(str(" \r\n").encode('utf-8'))
ser.close()

```
